tests/TestCase/ORM/TableTest.php

```diff
@@ -4237,21 +4252,25 @@ public function testBelongsToManyFluentInterface()
             ->setMethods(['_insert'])
             ->setConstructorArgs([['table' => 'authors', 'connection' => $this->connection]])
             ->getMock();
-
-        $authors->belongsToMany('articles')
-            ->setForeignKey('author_id')
-            ->setName('Articles')
-            ->setDependent(true)
-            ->setTargetForeignKey('article_id')
-            ->setBindingKey('id')
-            ->setConditions([])
-            ->setFinder('list')
-            ->setProperty('authors')
-            ->setSource($authors)
-            ->setStrategy('select')
-            ->setSaveStrategy('append')
-            ->setThrough('author_articles')
-            ->setJoinType('inner');
+        try {
+            $authors->belongsToMany('articles')
+                ->setForeignKey('author_id')
+                ->setName('Articles')
+                ->setDependent(true)
+                ->setTargetForeignKey('article_id')
+                ->setBindingKey('id')
+                ->setConditions([])
+                ->setFinder('list')
+                ->setProperty('authors')
+                ->setSource($authors)
+                ->setStrategy('select')
+                ->setSaveStrategy('append')
+                ->setThrough('author_articles')
+                ->setJoinType('inner');
+        } catch (\BadMethodCallException $e) {
+            $this->fail('Method chaining should be ok');
+        }
```

Comment:
It would work also but an error would be reporting instead of a falling test so I thought  it was  cleaner